src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -543,6 +547,58 @@ annotationExpression
         {#annotationExpression = #(#[EXPR,"EXPR"],#annotationExpression);}
     ;
 
+
+recordDefinition![AST modifiers]
+    :   r:"record" id:id
+        (tp:typeParameters)?
+        rc:recordComponentsList
+        ic:implementsClause
+        rb:recordBodyDeclaration
+        {#recordDefinition = #(#[RECORD_DEF, "RECORD_DEF"],
+                              modifiers, r, id, rc);}
+    ;
+
+// Build no AST for other record rules until
+// https://github.com/checkstyle/checkstyle/issues/8267
+recordComponentsList!
+    :   LPAREN recordComponents RPAREN
+    ;
+
+recordComponents!
+    // Taken from parameterDeclarationList
+    :   (   ( recordComponent )=> recordComponent
+            ( options {warnWhenFollowAmbig=false;} :
+                ( COMMA recordComponent ) => COMMA recordComponent )*
+            ( COMMA recordComponentVariableLength )?
+        |
+            recordComponentVariableLength
+        )?
+    ;
+
+recordComponentVariableLength!
+    :   parameterModifier t:variableLengthParameterTypeSpec ELLIPSIS id
+        declaratorBrackets[#t]
+    ;
+
+recordComponent!
+    :   annotations typeSpec[false] id
+    ;
+
+recordBodyDeclaration!
+    :   LCURLY
+        (   (compactConstructorDeclaration)=> compactConstructorDeclaration
+        |   field
+        |   SEMI
```

Comment:
http://cr.openjdk.java.net/~gbierman/jep359/jep359-20191031/specs/records-jls.html#jls-8.10.2

Reply:
I think we are covered with `field` and `SEMI` (I like this comment 😄) https://github.com/checkstyle/checkstyle/blob/f1ee17d45801f57c4a20a11d1311116c0e1d946c/src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g#L763-L768

This PR is parsing the [InputJava14Records](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java14/InputJava14Records.java) file which has records with fields, methods, etc. in their bodies.

Reply:
>Is there no ClassBodyDeclaration because of limited record support?

That was my original intention, but I don't see why we shouldn't go ahead and add it now.

Reply:
I can't provide recommendation on best way to go. If you believe this is parsing correctly or will be changed in full support, then you can mark this as resolved. I just noticed the disparity and wanted to make sure I was missing something or it was going to be resolved.

Reply:
This parses everything I've thrown at it, and all the records files in JDK14 repo.  The only changes I plan to make are to build the AST for it in #8267.  I'm going to mark it as resolved.